### PR TITLE
Allow custom user and port for builder host

### PR DESCRIPTION
**Pull request by mlitwiniuk**
When `ssh` options are set, they overwrite custom username and password passed via ssh builder uri. Passing part of uri for sshkit is fine, as it then properly extracts username and password and forwards it as `host.ssh_options` (in which case it's no longer empty)

### CODE REVIEW

File: lib/kamal/cli/build.rb [hunk 1]
```diff
@@ -115,7 +115,7 @@ def connect_to_remote_host(remote_host)
       remote_uri = URI.parse(remote_host)
       if remote_uri.scheme == "ssh"
         options = { user: remote_uri.user, port: remote_uri.port }.compact
-        on(remote_uri.host, options) do
+        on(remote_host.gsub('ssh://', ''), options) do
```
Comment (djmb) [inline lib/kamal/cli/build.rb:118]: I think we could do:

```ruby
host = SSHKit::Host.new(
  hostname: remote_uri.host,
  ssh_options: { user: remote_uri.user, port: remote_uri.port }
)
on(host) do
  execute "true"
end
```

which would avoid needing to alter the options in sshkit_with_ext.rb

Comment (mlitwiniuk) [reply lib/kamal/cli/build.rb]: Good catch, I've updated PR accordingly

**Comment by mlitwiniuk** [comment]
Yep, you're right. Closing

**Comment by djmb** [comment]
@mlitwiniuk - I was just referring to this line when I said we don't need it anymore. I think the rest of the PR is still valid!

```
host.ssh_options = (host.ssh_options || {}).merge({ port: host.port, user: host.user }.compact)
```

**Comment by djmb** [comment]
Going to re-open as I think we want this change!

**Comment by laptopmutia** [comment]
is it possible to add custom IdentityFile too ?

**Pull request merged**